src/index.js

```diff
@@ -5,46 +5,56 @@ import kafka from 'no-kafka';
 import NULL_LOGGER from 'null-logger';
 import Task from './task';
 import Registry from './registry';
-import Runner from './runner';
-import Admin from './admin';
-import Producer from './producer';
+import runner from './base/runner';
+import admin from './base/admin';
+import producer from './base/producer';
 import Config from './config';
 
-import type { Callback, Configuration, KafkaCompression } from '../types';
-
-const Steveo = (configuration: Configuration, logger: Object = NULL_LOGGER) => () => {
-  const registry = Registry();
-  const config = new Config(configuration);
-  let getTopicName = null;
-
-  const task = (topic: string, callBack: Callback) => {
-    const producer = Producer(config, registry, logger);
+import type { ITask, Configuration, Callback, Logger, ISteveo, IRegistry, IEvent, IAdmin } from '../types';
+
+class Steveo implements ISteveo {
+  config: Configuration;
+  logger: Logger;
+  registry: IRegistry;
+  getTopicName: Callback;
+  admin: IAdmin;
+  events: IEvent;
+
+  constructor(configuration: Configuration, logger :Logger = NULL_LOGGER) {
+    this.logger = logger;
+    this.registry = new Registry();
+    this.config = new Config(configuration);
```

Comment:
Am I correct in my read that if we wanted to publish to both Kafka and SQS in the one codebase, we'd have something like:

```
const Steveo = require('steveo');

const kafkaSteveo = new Steveo({engine: 'kafka'});
const sqsSteveo = new Steveo({enigne: 'sqs'});
```

If so, would we be better off just separating the packages altogether?

```
const KafkaSteveo = require('steveo-kafka');
const SqsSteveo = require('steveo-sqs');

const kafkaSteveo = new KafkaSteveo();
const sqsSteveo = new SqsSteveo();
```

We could still share the flow types between the two packages, meaning their APIs would remain equivalent since they both satisfy the same interfaces. I suspect it would simplify the code significantly.

Project layout would become something like:

```
tree steveo/

steveo
├── kafka
│   ├── index.js
│   ├── package.json
│   └── types.js -> ../types.js
├── sqs
│   ├── index.js
│   ├── package.json
│   └── types.js -> ../types.js
└── types.js
```